core/src/com/unciv/logic/GameInfo.kt

```diff
@@ -346,6 +346,12 @@ class GameInfo {
         }
 
         for (city in civilizations.asSequence().flatMap { it.cities.asSequence() }) {
+            // Temple was replaced by Amphitheater in 3.15.5. For backwards compatibility, we
+            // replace existing temples with amphitheaters. This replacement should be removed
+            // when temples are reintroduced as faith buildings.
+            if (!ruleSet.buildings.containsKey("Temple"))
```

Comment:
I think this 'if' should be part of the 'rename building' function

Reply:
Agreed. I didn't change it at first because it wasn't reflected by the name of the function, but I've changed the name to match its new semantics.